kong/resource_kong_plugin.go

```diff
@@ -191,3 +205,23 @@ func pluginConfigJsonToString(data map[string]interface{}) string {
 
 	return string(rawJson)
 }
+
+func findPlugin(
+	pluginClient *gokong.PluginClient, name string, consumerId *gokong.Id, routeId *gokong.Id, serviceId *gokong.Id,
+) (*gokong.Plugin, error) {
+	// size is just how many plugins per request (1000 is the max)
+	// but List will fetch all the pages so all the plugins
+	dbPlugins, err := pluginClient.List(&gokong.PluginQueryString{Size: 1000})
+	if err != nil {
+		return nil, fmt.Errorf("could not read existing plugins: %w", err)
+	}
+	for _, p := range dbPlugins {
+		if p.Name == name &&
+			gokong.IdToString(p.ConsumerId) == gokong.IdToString(consumerId) &&
+			gokong.IdToString(p.RouteId) == gokong.IdToString(routeId) &&
+			gokong.IdToString(p.ServiceId) == gokong.IdToString(serviceId) {
```

Comment:
I'd write a little function outside of this loop that takes `p` and checks all that.

Reply:
I prefer not TBH, I'd create it if it's useful elsewhere but meanwhile I don't really like to create an helper just for that in which you need to pass 5 parameters just for one statement.